data-prepper-plugins/otel-trace-raw-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepper.java

```diff
@@ -122,6 +131,9 @@ private void processRawSpan(final RawSpan rawSpan) {
     }
 
     private void processRootRawSpan(final RawSpan rawSpan) {
+        // TODO: flush descendants here to get rid of DelayQueue
+        // TODO: safe-guard against null traceGroup for rootSpan?
+        traceIdTraceGroupCache.put(rawSpan.getTraceId(), rawSpan.getTraceGroup());
```

Comment:
Just curious - does putting an item in the cache that already exists extend its TTL?

Reply:
Yes. According to the docs of expireAfterWrite: 
```
Specifies that each entry should be automatically removed from the cache once a fixed duration
has elapsed after the entry's creation, or the most recent replacement of its value.
```